feat(formEditor): track `diagramOpen`

smbea opened the pull request:
Closes #2405 

- [x]  Diagram open is tracked for forms similar to our other editors, including the engine profile, if set
- [x]  Telemetry documentation is updated accordingly

barmac commented:
I've tested this and it seems the engine profile is always an empty object, regardless if the form contains the information or not. Let's discuss it later today.

barmac commented:
https://user-images.githubusercontent.com/28307541/133238019-c084491d-74a3-40ee-92d0-3f89dc84b7ba.mov

Short footage of how I tested this.

barmac commented on client/src/plugins/usage-statistics/event-handlers/DiagramOpenEventHandler.js:
```diff
@@ -68,6 +69,18 @@ export default class DiagramOpenEventHandler extends BaseEventHandler {
+        tab
+      } = context;
+
+      const {
+        schema
+      } = tab;
```
Are you sure schema is passed with tab?

barmac commented on client/src/plugins/usage-statistics/event-handlers/DiagramOpenEventHandler.js:
```diff
@@ -136,6 +149,19 @@ export default class DiagramOpenEventHandler extends BaseEventHandler {
 
+  onFormOpened = async (schema, context = {}) => {
+
+    var engineProfile = { } ;
```
var -> const / let

barmac replied:
Also, I think we should not send an empty object as the engine profile.

barmac commented on client/src/plugins/usage-statistics/event-handlers/__tests__/DiagramOpenEventHandlerSpec.js:
```diff
@@ -1159,6 +1206,68 @@ describe('<DiagramOpenEventHandler>', () => {
+
+      const onSend = sinon.spy();
+
+      const tab = createTab({
+        type: 'form',
+        schema: engineProfilePlatform
+      });
```
Please keep the tab type as is. The tab should have a file property which has contents of type String. So my tab can look like:
```js
const tab = {
  file: {
    contents: '{"key": "value"}'
}
```

barmac replied:
Also, let's use real-life Camunda Forms files here.

barmac reviewed (changes requested):
See comments and the footage.

smbea commented:
Changes overview:
* Fixed the way the tab was being created in the tests and adjusted subscribe on `DiagramOpenEventHandler` to match the proper structure
* Doesn't send anything if file contents are broken (follows `bpmnEditor` behaviour)
* Omits `engineProfile` if not set

barmac commented:
Looks good to me. We are almost there, just one small thing. In #2405, the first requirement states that we want to track the execution platform version if available:

> Diagram open is tracked for forms similar to our other editors, including the engine profile (name, __version__), if set

smbea commented on client/src/plugins/usage-statistics/event-handlers/DiagramOpenEventHandler.js:
```diff
@@ -136,6 +150,43 @@ export default class DiagramOpenEventHandler extends BaseEventHandler {
+          executionPlatformVersion
+        } = schema;
+
+        if (executionPlatform) {
+
+          const engineProfile = executionPlatformVersion ?
+            { executionPlatform: executionPlatform, executionPlatformVersion: executionPlatformVersion }
+            : { executionPlatform: executionPlatform };
+
```
@barmac Added the executionPlatformVersion to the engineProfile, when set

barmac reviewed (approved):
Looks good. I was able to verify that it's working on my machine. Well done!

barmac commented on docs/telemetry/README.md:
```diff
@@ -199,7 +202,8 @@ If it is set in the diagram, we also add target engine profile information:
   "engineProfile": {
-    "executionPlatform": "Camunda Cloud"
+    "executionPlatform": "Camunda Cloud",
+    "executionPlatformVersion": "1.1"
```
This was added to a wrong event. Should be added after line 65.

smbea replied:
Thank you for the heads up! I moved this to the proper place.
